modules/renter/shardreader.go

```diff
@@ -0,0 +1,34 @@
+package renter
+
+import (
+	"io"
+)
+
+// shardReader is a helper struct that can read data into shards of
+// modules.SectorSize instead of whole byte slices.
+type shardReader struct {
+	r io.ReaderAt
+}
+
+// NewShardReader creates a new shardReader from an object that implements the
+// ReaderAt interface.
+func NewShardReader(r io.ReaderAt) *shardReader {
+	return &shardReader{
+		r: r,
+	}
+}
+
+// ReadAt reads data into a slice of shards from a certain offset.
+func (sr *shardReader) ReadAt(d [][]byte, offset int64) (int, error) {
```

Comment:
isn't this the wrong type signature?

Reply:
Yeah but you can't really create a 100% correct signature if you want to read into a slice of slices.